numpy/lib/function_base.py

```diff
@@ -1299,7 +1299,6 @@ def piecewise(x, condlist, funclist, *args, **kw):
                                    isinstance(condlist[0], ndarray))):
         condlist = [condlist]
     condlist = array(condlist, dtype=bool)
-    n = len(condlist)
     # This is a hack to work around problems with NumPy's
     #  handling of 0-d arrays and boolean indexing with
     #  numpy.bool_ scalars
```

Comment:
Why not? The text of the hack suggests that it is needed because of something like `arr[True]`? That works now. So I think you should be able to remove it (of course it should also simplify/need modification of some later code).